ogcapi-stable/ogcapi-features-core/src/main/java/de/ii/ldproxy/ogcapi/features/core/app/FeaturesCoreDataHydrator.java

```diff
@@ -241,4 +254,84 @@ private boolean hasMissingBboxes(Map<String, FeatureTypeConfigurationOgcApi> fea
                       })
                       .collect(ImmutableMap.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
     }
+
+    private boolean hasMissingIntervals(Map<String, FeatureTypeConfigurationOgcApi> featureTypes) {
+        return featureTypes
+                .entrySet()
+                .stream()
+                .anyMatch(entry -> entry.getValue()
+                        .getExtent()
+                        .isPresent() &&
+                        entry.getValue()
+                                .getExtent()
+                                .get()
+                                .getTemporalComputed());
+    }
+
+    private ImmutableMap<String, FeatureTypeConfigurationOgcApi> computeMissingIntervals(OgcApiDataV2 apiData) {
+        return apiData.getCollections()
+                .entrySet()
+                .stream()
+                .map(entry -> {
+                    FeatureProvider2 featureProvider = providers.getFeatureProvider(apiData, entry.getValue());
+
+                    if (entry.getValue()
+                            .getExtent()
+                            .isPresent() &&
+                            entry.getValue()
+                                    .getExtent()
+                                    .get()
+                                    .getTemporalComputed() &&
+                            featureProvider.supportsExtents()) {
+
+                        List<String> temporalQueryables = entry.getValue()
+                                .getExtension(FeaturesCoreConfiguration.class)
+                                .flatMap(FeaturesCoreConfiguration::getQueryables)
+                                .map(FeaturesCollectionQueryables::getTemporal)
+                                .orElse(ImmutableList.of());
+
+                        if (!temporalQueryables.isEmpty()) {
+                            Optional<Interval> interval;
+                            if (temporalQueryables.size() >= 2) {
+                                interval = featureProvider.extents()
+                                        .getTemporalExtent(entry.getValue().getId(), temporalQueryables.get(0), temporalQueryables.get(1));
+                            } else {
+                                interval = featureProvider.extents()
+                                        .getTemporalExtent(entry.getValue().getId(), temporalQueryables.get(0));
+                            }
+
+                            if (!interval.isPresent()) {
+                                return entry;
+                            }
+
+                            TemporalExtent temporalExtent = new ImmutableTemporalExtent.Builder()
+                                    .start(interval.get().getStart().toEpochMilli())
+                                    .end(interval.get().getEnd().toEpochMilli())
```

Comment:
This line throws an exception when the end of the interval was set to `Instant.MAX` because the Sql-Query returned `null`. To fix it, only set the end of `TemporalExtent` if not `interval.isUnboundedEnd()`.